### ISSUE TITLE
Tag GHCJS?

### ISSUE THREAD

**Issue by danse** [ghcjs/ghcjs#444]
I see a lot of developers trying to test GHCJS and building something on top of it, it is also starting to have some commercial applications. What would you think about using tags on this repository, in order to have easier references that can be depended on? Stack seem to allow to depend on a Git tag quite easily, and tagging should involve a minimal overhead from your side. Do you see any drawback in this?

**Comment by bergey** [comment]
+1

**Comment by pavelkogan** [comment]
+1

**Comment by spl** [comment]
Tags also make it easier to do and see releases on GitHub.

**Comment by 3noch** [comment]
+1

**Comment by danse** [comment]
What are [the versions used by Slack](http://docs.haskellstack.org/en/stable/ghcjs.html) referring to? Maybe this is already tagged, somehow

**Comment by bergey** [comment]
Luite builds [nightly tarballs](http://ghcjs.luite.com/), and it's quite easy now to have stack build with any one of those.  (Particularly since https://github.com/commercialhaskell/stack/issues/1622 ).  See for example https://github.com/bergey/gooey/blob/master/diagrams-minimal/stack.yaml

I keep meaning to update the Stack docs to reflect this.

**Comment by danse** [comment]
Cool, i wasn't aware of that list of nightly builds. That seems a sufficient way to refer to specific versions, until GHCJS will be published on Hackage. Given that list of builds, which is already integrated with Stack, i think that this issue could be closed.

**Comment by bergey** [comment]
@danse Glad I could help.  Do you have time to update the Stack docs to include this option?

**Comment by danse** [comment]
Don't they [already mention](http://docs.haskellstack.org/en/stable/ghcjs.html) how to point to a specific version of GHCJS? I was just missing information about what those releases were referring to. So what would you want to add to the docs, just something like *the GHCJS version used refers to nightly builds stored by Luite here: ...*?

**Comment by bergey** [comment]
Yes, I think the stack should mention the nightly tarballs.  As they are, it makes me think the 3 examples given are the only options.

**Comment by danse** [comment]
There is something unclear to me:

- when using a custom location, what is the role of the string passed as source? Does it even matter?
- when not providing a location, how is it derived from the source? Is the version part just ignored?

These questions raise from noticing that the sources include a version, for example as in `ghcjs-0.2.0.20151222`, while the nightly builds are just served by date as in `master-20151231`. Thus people should guess or know or find out the version corresponding to a given build.

It is probably to avoid this confusion, that the docs provide as examples pointers to `nrolland`'s fork of GHCJS, which has tags.

I could mention that the version can be found in the cabal file for a given archive, but i wonder wether that part of the source string is used at all. In fact, used this way versions are pointless. This makes me realise that having nightly builds does not solve the problem of versioning. It is more or less the equivalent of using Git commits as versions. Having tags and releases would still be useful

**Comment by bergey** [comment]
I believe that stack uses the argument to `compiler` (and to `source`) as the name of the directory where it unpacks the compiler.  For me, a subdirectory of `~/.stack/programs/x86_64-linux/`.  So it's important to use different compiler names for different tarballs, but I don't think the format of the string matters for GHCJS.

It's true that the date of a nightly build does not provide the same information as a semantic version number.  I think that's just the price of using a compiler with no releases yet.  The dates are still more human-readable than git commit hashes, and make it easier to pick a policy like "use the tarball from the first day of the current month; update our builds once a month."

Personally, I prefer the predictable frequency of luite's tarballs to the unknown time between nrollands.  YMMV.

**Comment by 3noch** [comment]
> It is probably to avoid this confusion, that the docs provide as examples pointers to nrolland's fork of GHCJS, which has tags.

Yes, luite's tarball names don't specify all the information that stack needs, so you have to supply that yourself in the `compiler` field. Stack basically trusts you to tell it the truth. If you're wrong, it may or may not cause problems (like telling it that GHCJS was built with GHC 7.10.3 when it's actually using 7.10.2 or something).

**Comment by danse** [comment]
Alright but my point is that, differently from what i stated above, nightly builds are *not* a sufficient way to refer versions, so not enough to close this issue

**Comment by 3noch** [comment]
Actually, yes. I agree with that completely!

**Comment by hanshoglund** [comment]
Cross-referencing #481
